### ISSUE TITLE
Make tests for the invoke module more stringent

### ISSUE THREAD

**Issue by gkapfham** [GatorEducator/gatorgrader#56]
Current the tests in the `test_invoke` suite are achieving full statement coverage. However, the tests do not have stringent checks on the status of the function under test, like `invoke.invoke_all_fragment_checks`. These tests should actually inspect the results of the report that is populated after performing the check and make sure that it contains the expected results.

**Comment by gkapfham** [comment]
Note that issue may be less of a concern once we go to a linting interface, as described in Issue #175.

**Comment by sjones-gh** [comment]
Think this would be a good issue for me to start on? I'm not sure if the linting interface lessened its importance, though.

**Comment by Michionlion** [comment]
I think this is definitely still something that needs to be fixed, although depending on your confidence in understanding of the system you may want to start with a simpler issue -- this one will require some knowledge of how we create a `report` dictionary, and what the contents of that dictionary should be for various invocations of the check functions. Note that some individual checks also may have test suites attached, and we want to try to multiplex our testing suites (have different inputs for as many situations as possible); thus, in adding to the `test_invoke` suite we'll want to make sure we're using different values and possibly triggering different edge cases that the individual check tests if possible.

**Comment by sjones-gh** [comment]
Yes, good point. I wouldn't want to make any duplicates or unnecessary tests. What issue would be a `good-first-issue` then?

**Comment by gkapfham** [comment]
Hi @sjones-gh, thanks for your interest in the project! Let us know if you have questions about picking an issue that you can start to work on as we are all glad to help.
